File: LeetCode/LeetCode 1313. 解压缩编码列表.py
class Solution:
    def decompressRLElist(self, nums: List[int]) -> List[int]:
        n = len(nums)
        ans = []

        for i in range(n):
            if i * 2 >= n:
                break
            x, y = nums[2 * i], nums[2 * i + 1]
            cur = [y] * x
            ans += cur
        
        return ans

File: LeetCode/LeetCode 1313. 解压缩编码列表.cpp
class Solution {
public:
    vector<int> decompressRLElist(vector<int>& nums) {
        int n = nums.size();
        vector <int> ans;
        for(int i = 0; i < n; i++) {
            if(2 * i >= n) break;
            int x = nums[2 * i], y = nums[2 * i + 1];
            for(int j = 1; j <= x; j++) {
                ans.push_back(y);
            }
        }
        return ans;
    }
};